ha/k8s_setup/ha_setup.py

```diff
@@ -261,7 +261,10 @@ def process(self):
             # TODO: Verify whether these newly added config is avilable in the confstore or not
             with open(const.HA_CONFIG_FILE, 'w+') as conf_file:
                 yaml.dump(conf_file_dict, conf_file, default_flow_style=False)
-            self._confstore = ConfigManager.get_confstore()
+            # Note: kv_enable_batch is setting True to commit all the kv keys in 1 time for time optimization.
+            # if kv_enable_batch set to true then commit operation needs to run to put all the key values in store.
+            kv_enable_batch_put = True
```

Comment:
Why not have separate functions, `_confstore::commit()` and `_confstore::commit_batch()` anyway you are setting `kv_enable_batch_put` to `True` explicitly?

Reply:
if 'kv_enable_batch_put' you set to False you don't need to call commit() in fact the commit() is only for batch operation.
I have added check in commit function as it should be called only in batch operation.

Reply:
Okay, so you mean there are some events which are saved in persistent store(Consul) and some not?
In any case, the way commit is invoked, it doesn't really seem to be related to `kv_enable_batch_put` or batching. I think commit can happen in a batch or not. It depends on the use case. That's why these things need to be explicit.
From this, I see that there can be multiple versions of commit, committing to a persistent store or not. This needs to be classified per event type and not really with respect to commit type.

Reply:
will fix in: https://jts.seagate.com/browse/CORTX-31319